src/bp/core/services/dialog/flow/utils.ts

```diff
@@ -0,0 +1,184 @@
+class TreeNode<T> {
```

Comment:
This whole file looks pretty generic, what do you think about moving it to `common/treeSearch.ts` so we can use it in the frontend, backend and modules ? Maybe we can find other uses for it elsewhere :)